.circleci/config.yml

```diff
@@ -161,6 +167,86 @@ install_doc_push_script: &install_doc_push_script
       echo "Skipping push due to dry_run"
     fi
 
+    popd
+
+    # ======================== Building PyTorch C++ API Docs ========================
+
+    echo "Building PyTorch C++ API docs..."
+
+    # Clone the cppdocs repo
+    rm -rf cppdocs
+    set +x
+    git clone https://pytorchbot:${GITHUB_PYTORCHBOT_TOKEN}@github.com/pytorch/cppdocs
+    set -x
+
+    set -ex
+
+    sudo apt-get -y install doxygen
+
+    # Generate ATen files
+    pushd "\${pt_checkout}"
+    pip install -r requirements.txt
+    time GEN_TO_SOURCE=1 python aten/src/ATen/gen.py \
+      -s aten/src/ATen \
+      -d build/aten/src/ATen \
+      aten/src/ATen/Declarations.cwrap \
+      aten/src/THNN/generic/THNN.h \
+      aten/src/THCUNN/generic/THCUNN.h \
+      aten/src/ATen/nn.yaml \
+      aten/src/ATen/native/native_functions.yaml
+
+    # Copy some required files
+    cp aten/src/ATen/common_with_cwrap.py tools/shared/cwrap_common.py
+    cp torch/_utils_internal.py tools/shared
+
+    # Generate PyTorch files
+    time python tools/setup_helpers/generate_code.py \
+      --declarations-path build/aten/src/ATen/Declarations.yaml \
+      --nn-path aten/src/
+
+    # Build the docs
+    pushd docs/cpp
+    pip install breathe==4.11.1 bs4 lxml six
+    pip install --no-cache-dir -e "git+https://github.com/pytorch/pytorch_sphinx_theme.git#egg=pytorch_sphinx_theme"
+    pip install exhale>=0.2.1
+    pip install sphinx==1.8.5
+    # Uncomment once it is fixed
+    # pip install -r requirements.txt
+    time make VERBOSE=1 html -j
+
+    popd
+    popd
+
+    pushd cppdocs
+
+    # Purge everything with some exceptions
+    mkdir /tmp/cppdocs-sync
+    mv _config.yml README.md /tmp/cppdocs-sync/
+    rm -rf *
+
+    # Copy over all the newly generated HTML
+    cp -r "\${pt_checkout}"/docs/cpp/build/html/* .
+
+    # Copy back _config.yml
+    rm -rf _config.yml
+    mv /tmp/cppdocs-sync/* .
+
+    # Make a new commit
+    git add . || true
+    git status
+    git config user.email "soumith+bot@pytorch.org"
+    git config user.name "pytorchbot"
+    # If there aren't changes, don't make a commit; push is no-op
+    git commit -m "Automatic sync on $(date)" || true
+    git status
+
+    if [ "\$dry_run" = false ]; then
```

Comment:
I'm a little concerned about the single `=`: it looks like it works, and it was in the code before, but it would be better to change it to `==` so no one reading this code gets confused